apps/challenges/aws_utils.py

```diff
@@ -1294,6 +1294,341 @@ def delete_log_group(log_group_name):
+        )
+
+        client = get_boto3_client("ec2", challenge_aws_keys)
+
+        client.revoke_security_group_ingress(
+            GroupId=challenge_evaluation_cluster.efs_security_group_id,
+            IpPermissions=[
+                {
+                    "FromPort": 2049,
+                    "IpProtocol": "tcp",
+                    "IpRanges": [
+                        {
+                            "CidrIp": challenge_obj.vpc_cidr,
+                        },
+                    ],
+                    "ToPort": 2049,
+                }
+            ],
+        )
+
+        # Delete Security Group
+        client.delete_security_group(
+            GroupName="evalai-code-upload-challenge-efs-{}".format(
+                environment_suffix
+            ),
+            GroupId=challenge_evaluation_cluster.efs_security_group_id,
+        )
+
+        client.delete_security_group(
+            GroupName="EvalAI code upload challenge",
+            GroupId=challenge_evaluation_cluster.security_group_id,
+        )
+
+        # Disassociate route table with subnets
+        route_table_association_ids = (
+            challenge_evaluation_cluster.route_table_association_ids
+        )
+        if route_table_association_ids:
+            for route_table_association_id in route_table_association_ids:
+                client.disassociate_route_table(
+                    AssociationId=route_table_association_id,
+                )
+
+        # Delete subnets
+        subnet_ids = [
+            challenge_evaluation_cluster.subnet_1_id,
+            challenge_evaluation_cluster.subnet_2_id,
+        ]
+        for subnet_id in subnet_ids:
+            client.delete_subnet(SubnetId=subnet_id)
+
+        # Delete and detach route table
+        client.delete_route(
+            DestinationCidrBlock="0.0.0.0/0",
+            RouteTableId=challenge_evaluation_cluster.route_table_id,
+        )
+        client.delete_route_table(
+            RouteTableId=challenge_evaluation_cluster.route_table_id
+        )
+
+        client.detach_internet_gateway(
+            InternetGatewayId=challenge_evaluation_cluster.internet_gateway_id,
+            VpcId=challenge_evaluation_cluster.vpc_id,
+        )
+        client.delete_internet_gateway(
+            InternetGatewayId=challenge_evaluation_cluster.internet_gateway_id
+        )
+
+        client.delete_vpc(VpcId=challenge_evaluation_cluster.vpc_id)
+
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+
+def delete_setup_eks_cluster(challenge):
+    """
+    Delets EKS and NodeGroup ARN roles
+
+    Arguments:
+        instance {<class 'django.db.models.query.QuerySet'>} -- instance of the model
+    """
+    from .models import ChallengeEvaluationCluster
+    from .serializers import ChallengeEvaluationClusterSerializer
+    from .utils import get_aws_credentials_for_challenge
+
+    # Delete eks cluster vpc and subnets
+    delete_eks_cluster_subnets(challenge)
```

Comment:
What if this fails? Need to capture this and trigger a failure in this method as well @savish28 
cc: @Ram81